install/mac.rst

```diff
@@ -162,39 +165,123 @@ different between the two:
+
+.. code-block:: console
+
+  . ~/mpfenv/bin/activate
+
+Note that the first character is a period, followed by a space, then the path
+to your virtual environment and "/bin/activate".
+
+.. note::
+
+  You may want to write this step down, as you'll run it every time you open up
+  a terminal window to work on MPF*
+
+You'll know you're in the virtual environment because the console prompt will include
+the name of your venv in parenthesis.
+
+.. code-block:: console
+
+  My-Mac:~ python --version
+  Python 2.7.10
+  My-Mac:~ . ~/mpfenv/bin/activate
+
+  (mpfenv) My-Mac:~ python --version
+  Python 3.6.8
+  (mpfenv) My-Mac:~ 
+
+.. note::
+  
+   The python you used to create the virtual environment will now be the 
+   default python. Outside the virtual environment "python" is Python 2 and
+   you must type "python3" to use Python 3; inside the virtual environment,
+   you can use "python" to refer to Python 3.
+
 4. Install/upgrade some Python components
 -----------------------------------------
 
 Python includes a utility called "pip" which is the name of the Python Package
 Manager. Pip is used to install Python packages and applications from
 the web. (It's kind of like an app store for Python apps.)
 
-Due to a bug in versions of pip older than 9.0.2 on the Mac, we cannot update *pip*
-using *pip*. So the next step is to download and run a special Python script to install
-the latest version of pip.
+If you are not using a virtual environment and have both Python 2 and Python 3 
+on your system, you'll most likely need to use "pip3". Check your version to see:
 
-Update pip by running the following command:
+.. code-block:: console
+  
+  My-Mac:~ $ pip --version
+  pip 8.0.2 from /usr/bin/pip (python 2.7)
+  My-Mac:~ $ pip3 --version
+  pip 16.2.1 from /usr/bin/pip3 (python 3.6)
+
+If your "pip" is for Python 2, then you'll use "pip3" through the rest of this guide. 
+
+If you created a virtual environment using Python 3, then "pip" will be the same 
+as "pip3" and you can use them interchangably.
+
+.. code-block:: console
+  
+  (mpfenv) My-Mac:~ $ pip --version
+  pip 19.0.1 from ~/venv/lib/python3.6/site-packages/pip (python 3.6)
+  (mpfenv) My-Mac:~ $ pip3 --version
+  pip 19.0.1 from ~/venv/lib/python3.6/site-packages/pip (python 3.6)
+
+The versions of pip that come with Python aren't always the newest, so it's a
+good idea to update pip by running the following command:
 
 .. code-block:: console
 
-    curl -O https://bootstrap.pypa.io/get-pip.py
-    python3 get-pip.py
+  pip install --update pip
 
-The latest version of pip should now be installed (9.0.3 or newer).
+The latest version of pip should now be installed.
 
 Next, we need to install and update a few other python packages required to run mpf by
 running the following command:
 
-So next run the following command:
-
 .. code-block:: console
 
-    pip3 install setuptools cython==0.25.2 --upgrade
+    pip install --upgrade setuptools cython
```

Comment:
I guess we need "pip3 install -I Cython==0.27.3". See https://github.com/missionpinball/mpf-debian-installer/blob/dev/install-mc-dependencies#L11

Reply:
Are there concerns about future versions? I've got 0.29.3 in both my 3.6 and 3.7 versions and they work just fine.

Reply:
It really only depends on kivy. They document which versions will work